src/main/java/io/github/programminglife2016/pl1_2016/parser/database/SimpleDatabase.java

```diff
@@ -0,0 +1,421 @@
+                node = new Segment(segmentid, rs.getString("data"), rs.getInt("column_index"));
+                node.setXY(rs.getInt("positionx"), rs.getInt("positiony"));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        if (stmt != null) {
+            stmt.close();
+        }
+        return node;
+    }
+
+    /**
+     * Write a collection of segments into the database
+     *
+     * @param nodes the collection to write
+     * @throws SQLException thrown if SQL connection or query is not valid
+     */
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void writeSegments(NodeCollection nodes) throws SQLException {
+        clearTable(SEGMENT_TABLE);
+        writeLinks(nodes);
+        PreparedStatement stmt = null;
+        String query = "INSERT INTO " + SEGMENT_TABLE
+                + "(segment_id, data, column_index, positionx, positiony) VALUES"
+                + "(?,?,?,?,?)";
+        try {
+            stmt = connection.prepareStatement(query);
+            for (Node node : nodes.values()) {
+                stmt.setInt(1, node.getId());
+                stmt.setString(2, node.getData());
+                stmt.setInt(3, node.getColumn());
+                stmt.setInt(4, node.getX());
+                stmt.setInt(5, node.getY());
+
+                stmt.executeUpdate();
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            if (stmt != null) {
+                stmt.close();
+            }
+        }
+    }
+
+    private void clearTable(String tableName) {
+        Statement stmt;
+        try {
+            stmt = connection.createStatement();
+            String query = "DELETE FROM " + tableName;
+            stmt.execute(query);
+            if (stmt == null) {
+                stmt.close();
+            }
+        } catch (SQLException s) {
+            s.printStackTrace();
+        }
+    }
+
+    private void writeLinks(NodeCollection nodes) throws SQLException {
+        clearTable(LINK_TABLE);
+        PreparedStatement stmt = null;
+        String query = "INSERT INTO " + LINK_TABLE
+                + "(from_id, to_id) VALUES"
+                + "(?,?)";
+        try {
+            stmt = connection.prepareStatement(query);
+            for (Node node : nodes.values()) {
+                for (Node link : node.getLinks()) {
+                    stmt.setInt(1, node.getId());
+                    stmt.setInt(2, link.getId());
+                    stmt.executeUpdate();
+                }
+
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            if (stmt != null) {
+                stmt.close();
+            }
+        }
+    }
+
+    /**
+     * Fetch all specimen from database
+     *
+     * @return the collection of specimen
+     * @throws SQLException thrown if SQL connection or query is not valid
```

Comment:
Can you not split this method into a few smaller ones instead of suppressing checkstyle